Nos/Models/CoreData/Event+CoreDataClass.swift

```diff
@@ -557,6 +558,9 @@ public class Event: NosManagedObject, VerifiableEvent {
         } else {
             let event = Event(context: context)
             event.identifier = id
+            context.perform {
```

Comment:
This looks like something that could have side-effects in other parts of the app. Mind if I ping @mplorentz so he takes a look at it?

Reply:
I moved it to avoid side-effects from other callers of this function.

Reply:
This does give me pause, because this function should only ever be called in a `context.perform` block already (but there's no way for you to know that @bryanmontz, it's just our convention) and I think I have seen nested context.perform blocks deadlock the app before. But this code has been on staging for some days and I haven't heard any reports of deadlock, so maybe it's fine? I guess I would rather it was reverted if you have the time. 

In general all the instance functions on NSManagedObjects are context agnostic and it's up to the caller to ensure that they are executed on the thread that Core Data wants. In the case of the view context it wants the main thread, and for all other contexts it wants to be in the context.perform block.